app/models/edition.rb

```diff
@@ -66,6 +66,31 @@ class Edition < ApplicationRecord
 
   MINIMUM_SCHEDULING_TIME = { minutes: 15 }.freeze
 
+  scope :find_current, ->(id: nil, document: nil) do
+    find_by = {}.tap do |criteria|
+      criteria[:id] = id if id
+
+      if document
+        content_id, locale = document.split(":")
+        criteria[:documents] = { content_id: content_id, locale: locale }
+      end
+    end
+
+    join_tables = %i[document revision status]
+    where(current: true)
+      .joins(join_tables)
+      .includes(join_tables)
+      .find_by!(find_by)
+  end
+
+  def self.find_and_lock_current(*args)
+    transaction do
+      lock.find_current(*args).tap do |edition|
```

Comment:
Very minor, but I think you can do `tap(&block)` here if `&block` is also an arg to the main method.

Reply:
That's cool - thanks, I've tried that out. 

Pro's are it can be nice and succinct - now fits on one line.
Con's seem to be that it's upped the magic of it and has probably pushed it past being able to read the method and now how it should be used.

Reply:
Fair enough :-).